core/arch/interp.c

```diff
@@ -3744,6 +3744,21 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             break;
         }
 
+#ifdef X86
+        if (my_dcontext != NULL && instr_is_syscall(bb->instr) &&
+            instr_get_opcode(bb->instr) == OP_int &&
```

Comment:
Well, there is a good question that relates to my broader question of how to fix this case.

Should we relax the constraint in arch.c :
```
/* we assume only single method; else need multiple do_syscalls */
    ASSERT(new_method == get_syscall_method());
```